app/services/amr/csv_parser_and_upserter.rb

```diff
@@ -13,24 +13,17 @@ def perform
 
       amr_data_feed_import_log = AmrDataFeedImportLog.create(amr_data_feed_config_id: @config.id, file_name: @file_name, import_time: DateTime.now.utc)
 
-      @upserted_record_count = DataFeedUpserter.new(get_unique_array_of_readings, amr_data_feed_import_log.id).perform
+      amr_reading_data = CsvToAmrReadingData.new(@config, "#{@config.local_bucket_path}/#{@file_name}").perform
```

Comment:
Should we check the validity here?

Reply:
Yes, we now do check the validity and update the Amr Import Log if there's a problem